key_action_map.cpp

```diff
@@ -163,6 +168,77 @@ sf::Keyboard::Key get_stun_key(const key_action_map& m)
   return m.to_key(action_type::shoot_stun_rocket);
 }
 
+key_action_map load_kam(const std::string& filename)
+{
+  std::ifstream f(filename);
+  key_action_map m;
+  f >> m;
+  return m;
+}
+
+void save_to_file(const key_action_map& kam, const std::string& filename)
+{
+  std::ofstream file(filename);
+  file << kam;
+}
+
+sf::Keyboard::Key to_sfml_key(const std::string& s)
+{
+  if (s == "A") return sf::Keyboard::Key::A;
```

Comment:
Hi @EvoLandEco, indeed, this is a clumsy incomplete function. The C++20/C++24 equivalent of the R `eval` is called 'reflection'. Because we do not use C++20/C++24 (yet), this clumsy way is the only work forward.

There will be smarter ways (using a look-up `std::map`), which will be a future junior Issue :-)

Reply:
Good to know that C++ finally has this quality of life feature, though I see many C++ programmers questioning the efficiency and safety issues of doing this

Reply:
Created #530

Reply:
And also: shouldn't it be `if (s == "3") return sf::Keyboard::Key::A;`?
In test.txt I see `3 turn_left`, but not `A turn_left`. I guess the `<<` operator overloaded for sfml::Keyboard will convert it to a digit which is the index of the key?

Reply:
> I guess the << operator overloaded for sfml::Keyboard will convert it to a digit which is the index of the key?

I guess it does. For now. Some future junior Issues (where one of write the tests) should fix that :-)